Extract message-ID into a separate column, save attachments to storage not db, added config file

dmcbrn opened the pull request:
Separate message-ID into it's own column to facilitate identifying emails.

In my scenario I have a webhook that will ping my server when the email failed and this gives me a quick way to find in the `emails_log` table which email has failed.

dmcbrn committed:
add separate column for messageId and index it

dmcbrn committed:
save messageId in a separate indexed column

dmcbrn committed:
Save attachments to `email_log_attachments` folder

Instead of saving the attachments to the database, save them to the `storage/email_log_attachments/message-id/` folder instead.

dmcbrn commented:
In commit [f39adf3](https://github.com/shvetsgroup/laravel-email-database-log/pull/13/commits/f39adf34a0e36445e28288f724aa7822fd7173fe) I've added saving attachments to the `storage/email_log_attachments` folder instead of saving them to the database.

dmcbrn committed:
Added missing path to storage class

dmcbrn committed:
Update README.md

baileylo commented on src/ShvetsGroup/LaravelEmailDatabaseLog/EmailLogger.php:
```diff
@@ -16,6 +17,15 @@ public function handle(MessageSending $event)
+        $attachments = [];
+        foreach ($message->getChildren() as $child) {
+            $attachmentPath = 'email_log_attachments/' . $messageId . '/' . $child->getFilename();
+            Storage::put($attachmentPath,$child->getBody());
```
There should be a space after the `,`. 

```php
Storage::put($attachmentPath, $child->getBody());
$attachments[] = $attachmentPath;
```

Also, it's better to use the array assignment operators rather than `array_push` when adding a single item to an array. I'd say this amount to a micro optimization, as the over head is the function over head, but it is stated in the php docs.

dmcbrn replied:
Updated.

baileylo commented on src/database/migrations/2018_07_05_131300_add_message_id_column.php:
```diff
@@ -0,0 +1,34 @@
+    public function up()
+    {
+        Schema::table('email_log', function (Blueprint $table) {
+            $table->string('messageId', 32)->nullable()->default(null);
```
I think `message_id` might be a better name. There is no set precedent for snake case vs camel case at the table level, but the table name being snake case I think it follows that the columns should be snake case.

dmcbrn replied:
Here is the migration code 

    up() {
       Schema::table('email_log', function (Blueprint $table) {
        $table->renameColumn('messageId', 'message_id');
      });
    }
    down() {
       Schema::table('email_log', function (Blueprint $table) {
        $table->renameColumn('message_id', 'messageId');
      });
    }


@baileylo Do you maybe know do I need to change the indexes also or would this suffice?

dmcbrn committed:
Added spaces after commas

dmcbrn committed:
Micro optimization of attachment handling

dmcbrn committed:
Created config file

Created a config file and a default option for where to save the attachments (folder name).

dmcbrn committed:
Added config file

dmcbrn committed:
Proper placement of config file

dmcbrn committed:
Wrong place for the config file

dmcbrn committed:
Again wrong place for config file

dmcbrn committed:
Again wrong place for config file

dmcbrn committed:
Updated paths

dmcbrn committed:
Renamed config file.

dmcbrn committed:
Updated path to config file

dmcbrn committed:
Fixed typo in syntax - starting <?php

dmcbrn committed:
Updated code to use config.email_log.folder

dmcbrn committed:
Created config file at better place in code

dmcbrn committed:
Delete old config file

dmcbrn committed:
Updated paths to new config file

dmcbrn commented:
A bit of a mess up there but the end result is config file which contains the name of the folder where the attachments will be saved.

Sorry for the mess I'll try not to update code directly on GitHub anymore :flushed:

dmcbrn committed:
Fixed publishing config file path

dmcbrn committed:
Added info on publishing config file

dmcbrn commented:
Seems like a dead repo.

Commit:
initial reorganization

neochief commented:
It’s not dead, I just waited until you finish :D

Commit:
using proper filepath for migration and config files

dmcbrn committed:
using new composer package name

Commit:
correcting namespaces

Commit:
fixing merge conflict

Commit:
readme file updates

Commit:
updating readme file with info on attachments folder

Commit:
save mail_driver to table for future handling of webhooks

Commit:
set up proper path to migrations

Commit:
disable saving attachments

Commit:
reintroduced saving of attachments

Commit:
use new EmailLog Eloquent model to create database entry

Commit:
list through attachments only if return child is of type Swift_embeddedFile or Swift_attachment

baileylo commented on src/EmailLogger.php:
```diff
@@ -20,9 +20,12 @@ public function handle(MessageSending $event)
-            Storage::put($attachmentPath, $child->getBody());
-            $attachments[] = $attachmentPath;
+            //docs for this below: http://phpdox.de/demo/Symfony2/classes/Swift_Mime_SimpleMimeEntity/getChildren.xhtml
+            if(in_array(get_class($child),['Swift_EmbeddedFile','Swift_Attachment'])) {
```
I don't know enough about this change, but am curious what this eliminates?

dmcbrn replied:
I had some issues that the object returned was of type Swift_MimePart which doesn't have a `getFilename()` method defined, so this is a workaround.

Didn't have the time to investigate into this properly. I think it would be the same if it was checking for only Swift_Attachment type.

neochief commented:
@dmcbrn Are you still interested in merging all of this into original repo?

baileylo commented:
@neochief Doesn't make sense after they removed you from the copyright, changed the namespaces, and composer package. Seems really weird.

dmcbrn commented:
@neochief @baileylo 

Hi, 

the thing is I use this in my production and plan to upgrade it to a bit more complete package. Since you haven't touched it in a while and didn't reply to my pull request I decided to make my own version of it. 

Regarding the copyright or anything else. It still says "my" package is fork of yours so I hoped that's how it's done. If I'm not approaching this properly please let me know. I'm rather new to this all and my intention is not to be unpolite or to steal credit for your part of work.

dbr0 committed:
reviewing of sent emails is now possible via /email-log routes which can be prefixed and middlewared by new config options

dmcbrn committed:
Updated readme file with instructions

dbr0 committed:
added missing routes file

dbr0 committed:
Merge branch 'master' of https://github.com/dmcbrn/laravel-email-database-log

Pulled latest readme file changes.

dbr0 committed:
added missing config for route prefixes

dbr0 committed:
implemented web middleware to routes

dmcbrn committed:
Minor updates to readme

dmcbrn committed:
Update README.md

dmcbrn committed:
Update README.md

dbr0 committed:
optimized query for listing emails

dbr0 committed:
Merge branch 'master' of https://github.com/dmcbrn/laravel-email-database-log

New readme file updated directly on GitHub.

dbr0 committed:
returning proper attachments, new values in require section of composer

dbr0 committed:
emails now have events associated with them which can be accessed through POST route - to be used with webhooks section in your mail sending service

dbr0 committed:
emails now handle webhooks request from mailgun properly and save the data

dbr0 committed:
list events when listing emails and enable timestamps on events

dbr0 committed:
fixed relations namespaces

dbr0 committed:
fixed relation keys

dbr0 committed:
fixed request namespace

dbr0 committed:
enable timestamps on event model

dbr0 committed:
refactored saving event code

dbr0 committed:
updated readme to note that the library can be used with mailgun webhooks

dbr0 committed:
abstracted Events so we can use multiple mail-webhook-services in the future

dbr0 committed:
abstracted Events so we can use multiple mail-webhook-services in the future - fixed abstract,public,namespace issues

kentdahl committed:
Publish config example did not work on case-sensitive filesystem.

kentdahl committed:
Typo in namespace for interfaces for events.

dmcbrn committed:
Merge pull request #2 from rubendahl/typo-namespace-interfaces

Typo in namespace for interfaces for events.

dmcbrn committed:
Merge pull request #1 from rubendahl/readme-publish-config-casesensitive-fix

Publish config example did not work on case-sensitive filesystem.

dbr0 committed:
properly handle listing of attachements - missing $key variable in foreach()

dbr0 committed:
Merge remote-tracking branch 'refs/remotes/origin/master'

dbr0 committed:
you can delete emails older than a date, including the events, and attachments

dbr0 committed:
you can delete emails older than a date, including the events, and attachments - 2/2 - missing updates to files

dbr0 committed:
you can use simple <like> search to search for emails containing provided <to> address and <subject>

dbr0 committed:
MailgunEvent saves proper messageId to EmailLogEvent

dbr0 committed:
relate emails and events through messageId

dbr0 committed:
response to Mailgun is changed to 200 when no email found on local server as the 400 will trigger Mailgun to try to retry the event

dmcbrn committed:
create Laravel Event on logging email

dmcbrn committed:
ommited @error syntax from view as only usable in 5.8.13 and requirements are ~5.5

dmcbrn committed:
readme file updated with instructions on how to use EmailLogged event

dmcbrn committed:
Merge branch 'master' of https://github.com/dmcbrn/laravel-email-database-log

arsenipachkovski committed:
chore: Update dependencies to laravel 6

arsenipachkovski committed:
fix: Remove useless string before php opening tag

dmcbrn committed:
Merge pull request #6 from ArseniPachkovski/master

chore: Update dependencies for Laravel 6

dmcbrn committed:
psr-0 to psr-4

dmcbrn committed:
how to use with Laravel 5

dmcbrn committed:
Laravel 7 support

dmcbrn committed:
Which release to use for which version of Laravel

tsekka committed:
Update composer.json

tsekka committed:
Use multiple middlewares

tsekka committed:
Fix event not firing

dmcbrn committed:
Laravel 8 support

dmcbrn committed:
readme update

dmcbrn committed:
Merge branch 'master' into master

dmcbrn committed:
Merge pull request #8 from tsekka/master

Laravel 8 only + add possibility to use multiple middlewares from env

dmcbrn committed:
minor code algnment

dmcbrn committed:
align attachments separator

dmcbrn committed:
slight change on how the attachments are fetched
